src/dataframerow/utils.jl

```diff
@@ -111,16 +112,16 @@ function row_group_slots(cols::Tuple{Vararg{AbstractVector}},
     @assert sz >= length(rhashes)
     szm1 = sz-1
     gslots = zeros(Int, sz)
-    # If missings are to be skipped, they will all go to group 1,
-    # which will be removed by group_rows
-    ngroups = skipmissing ? 1 : 0
+    # If missings are to be skipped, they will all go to group 0,
+    # which will be removed by functions down the stream
+    ngroups = 0
     @inbounds for i in eachindex(rhashes)
         # find the slot and group index for a row
         slotix = rhashes[i] & szm1 + 1
-        # Use 0 for non-missing values to catch bugs if group is not found
-        gix = skipmissing && missings[i] ? 1 : 0
+        # Use -1 for non-missing values to catch bugs if group is not found
+        gix = skipmissing && missings[i] ? 0 : -1
         probe = 0
-        # If skipmissing=true, assign rows containing at least one missing to group 1
+        # If skipmissing=true, assign rows containing at least one missing to group 0
```

Comment:
I do not understand this comment (especially given the earlier comment).
The code below seems to be executed if `skipmissing==false` or if `missings[i]==false` so this.
Is this related to the fact that we initialized `gslots` to `0`. But if this is the case then how did this work before as the condition and the initialization has not changed.

Reply:
The group index is always set to `gix`, which previously was `1` for missing values and is now `0`. `gslots` is something different: there a 0 indicates an unfilled slot in the hash table (it's not used for missing values).

Can you suggest a better wording?

Reply:
Again - I need to understand this code first 😭. If `skipmissing==true` and find missing then we set `gix=0` and later set `groups[i] = gix` to skip this group - right? If this is so I would just add something that group `0` is later skipped?

Reply:
Also maybe in https://github.com/JuliaData/DataFrames.jl/pull/2047/files#diff-223a9905c9dcda6f61927d162475af85L129 it would be better to write:
```
ngroups += 1
gix = ngroups
```
to make the code simpler? (this is a minor thing, but I prefer to have more readable code especially where it is complex)

Reply:
Yes, that's it. Feel free to suggest a wording. (Actually the comment is a bit redundant with the one at the top of the loop...)

I'm not sure `gix = ngroups += 1` is the most complex thing in this code :-). But why not divide it in two steps if you prefer...

Reply:
> (Actually the comment is a bit redundant with the one at the top of the loop...)

This is exactly what misguided me initially.

Reply:
```suggestion
        # If skipmissing=true, rows containing at least one missing get a group number 0 which is skipped later
```